citations-api/api/src/java/org/sakaiproject/citation/api/CitationCollectionOrder.java

```diff
@@ -0,0 +1,110 @@
+package org.sakaiproject.citation.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by nickwilson on 6/23/15.
+ */
+public class CitationCollectionOrder {
+
+	private String collectionId;
+	private String citationid;
+	private int location;
+	private SectionType sectiontype;
+	private String value;
+	private List<CitationCollectionOrder> children = new ArrayList<CitationCollectionOrder>();
+
+	public enum SectionType {
+		HEADING1, HEADING2, HEADING3, DESCRIPTION, CITATION
+	}
+
+	public CitationCollectionOrder() {
+	}
+
+	public CitationCollectionOrder(String collectionId, String citationid, int location, SectionType sectiontype, String value) {
+		this.collectionId = collectionId;
+		this.citationid = citationid;
+		this.location = location;
+		this.sectiontype = sectiontype;
+		this.value = value;
+	}
+
+	public CitationCollectionOrder(String collectionId, int location, SectionType sectiontype, String value) {
+		this.collectionId = collectionId;
+		this.location = location;
+		this.sectiontype = sectiontype;
+		this.value = value;
+	}
+
+	public List<CitationCollectionOrder> getChildren() {
+		return children;
+	}
+
+	public void setChildren(List<CitationCollectionOrder> children) {
+		this.children = children;
+	}
+
+	public void addChild(CitationCollectionOrder citationCollectionOrder) {
+		this.children.add(citationCollectionOrder);
+	}
+
+	public String getCitationid() {
+		return citationid;
+	}
+
+	public void setCitationid(String citationid) {
+		this.citationid = citationid;
+	}
+
+	public SectionType getSectiontype() {
+		return sectiontype;
+	}
+
+	public void setSectiontype(SectionType sectiontype) {
+		this.sectiontype = sectiontype;
+	}
+
+	public String getCollectionId() {
+		return collectionId;
+	}
+
+	public void setCollectionId(String collectionId) {
+		this.collectionId = collectionId;
+	}
+
+	public int getLocation() {
+		return location;
+	}
+
+	public void setLocation(int location) {
+		this.location = location;
+	}
+
+	public String getValue() {
+		return value;
+	}
+
+	public void setValue(String value) {
+		this.value = value;
+	}
+
+	public List<CitationCollectionOrder> flatten() {
```

Comment:
I'm not sure about this method, it means in the code you end up with 2 sorts of CitationCollectionOrder objects, ones that are hierarchical and ones that are in a flattened list and it's not easy to tell the 2 apart. It's not a show-stopper, just something that doesn't feel quite right.